src/components/TreeGrid.ts

```diff
@@ -46,13 +46,41 @@ const wrapContent =
     state: 'closed',
   });
 
+const hierarchyMapToRows = <T extends RowContents>(hierarchyMap: Map<number, T[]>): Row<T>[] => {
+  const traverseHierarchy = (parent: T, depth: number, previousRows: Row<T>[]): Row<T>[] => {
+    // does parent have children?
+    const children = hierarchyMap.get(parent.id) || [];
+    const parentRow: Row<T> = {
+      contents: parent,
+      depth,
+      isFetched: children.length > 0,
+      state: children.length > 0 ? 'open' : 'closed',
+    };
+    // recursively traverse hierarchy of all children
+    const childRows: Row<T>[] = children.flatMap((child) => traverseHierarchy(child, depth + 1, []));
+    return [...previousRows, parentRow, ...childRows];
+  };
+
+  // hierarchyMap assumes that the root is the last key
+  const rootID: number | undefined = Array.from(hierarchyMap.keys()).pop();
```

Comment:
Thoughts on calling this domainOptionRootID or rootID?

Reply:
the `Array.from ...` syntax comes from https://stackoverflow.com/a/30921449/13271387
Thoughts on this?

Reply:
This looks reasonable to me - But we may consider (in the future) making a change on the API to remove the need for this logic by returning more information about the root in the api response.